File: java/JavaEg/src/com/examples/WorldBank.java
package com.examples;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class WorldBank
{

    public static void main(String[] args) throws IOException
    {
        getWBByPage();
        getWBByCategory();
        getWBIndicator();
        getWBCategories();
        getWBCountryPage();
        getWBHistorical();
        getWBByUrl();

    }

    public static void constructUrl(String path) throws IOException
    {
        String _clientKey = "guest:guest";
        String base_url = "https://api.tradingeconomics.com";
        String auth;
        if (path.contains("?"))
            auth = base_url + path + "&c=" + _clientKey;
        else
            auth = base_url + path + "?c=" + _clientKey;

        URL obj = new URL(auth);
        HttpURLConnection con = (HttpURLConnection) obj.openConnection();

        BufferedReader in = new BufferedReader(
                new InputStreamReader(con.getInputStream()));
        String inputLine;
        StringBuffer response = new StringBuffer();

        while ((inputLine = in.readLine()) != null) {
            response.append(inputLine);

        }
        in.close();

        //print result
        System.out.println(response.toString());
    }

    public static  void getWBCategories() throws IOException
    {

        //set the path for the query
        String path = "/worldBank/categories";

        System.out.println("--------Get main categoriess--------");
        constructUrl(path);
    }

    public static void getWBByCategory() throws IOException {

        //put category name here
        String params = "Education";
        params = params.replaceAll("\\s","%20");
        //set the path for the query
        String path = "/worldBank/category" + "/" + params ;

        System.out.println("--------Get world bank data by category--------");
        constructUrl(path);

    }
    public static void getWBByPage() throws IOException {

        //put category name here
        String params = "Education";
        params = params.replaceAll("\\s","%20");
        //put page number here
        String page = "2";
        //set the path for the query
        String path = "/worldBank/category" + "/" + params + "/" + page;

        System.out.println("--------Get world bank data by category and page number--------");
        constructUrl(path);

    }

    public static void getWBIndicator() throws IOException {

        //put series code here
        String series_code = "fr.inr.rinr";
        series_code = series_code.replaceAll("\\s","%20");
        //set the path for the query
        String path = "/worldBank/indicator" + "?s=" + series_code;

        System.out.println("--------Get world bank data by specific indicator--------");
        constructUrl(path);

    }
    public static void getWBCountryPage() throws IOException {

        //put country name here
        String params = "portugal";
        params = params.replaceAll("\\s","%20");
        //put page number here
        String page = "2";
        //set the path for the query
        String path = "/worldBank/country" + "/" + params + "/" + page;

        System.out.println("--------Get world bank data by specific country and page--------");
        constructUrl(path);

    }

    public static void getWBByUrl() throws IOException {

        //put urlhere
        String url = "/united-states/real-interest-rate-percent-wb-data.html";
        url = url.replaceAll("\\s","%20");
        //set the path for the query
        String path = "/worldBank/indicator" + "?url=" + url ;

        System.out.println("--------Get world bank data by specific url--------");
        constructUrl(path);

    }

    public static void getWBHistorical() throws IOException {

        //put series code here
        String series_code = "usa.fr.inr.rinr";
        series_code = series_code.replaceAll("\\s","%20");
        //set the path for the query
        String path = "/worldBank/historical" + "?s=" + series_code ;

        System.out.println("--------Get world bank historical data by specific series code--------");
        constructUrl(path);

    }
}

File: python/examples-x/cli_terminal/ClassLibrary/world_bank.py
import tradingeconomics as te
import time

from .section import Section

class WorldBank():
    def __init__(self, key):
        self.key=key
        self.title = '******* Trading Economics - WorldBank  ********'
        self.section_name = 'WorldBank'
        
        self.dictionary={
                    '1':' - List of Main categories',
                    '2':' - Filtering by the main categories',
                    '3':' - Detailed information about specific indicator for all countries using a series code from the previous option',
                    '4':' - List of indicators available for a specific country',
                    '5':' - Detailed information just for a specific indicator and country by using series code',
                    '6':' - Detailed information just for a specific indicator and country by using url',
                    '7':' - Historical data for a specific indicator',
                    '8': ' - Back to Main Menu'
                    
                    }
    
        def select_function(self):
            section = Section()
            te.login(self.key)
            list_of_number = self.dictionary.keys()
            number = ''
            
            number = section.first_section(section_title=self.title,
                                    section_dictionary=self.dictionary,
                                    list_of_number = list_of_number)
            
            all_dict_values = list(self.dictionary.values())
            back_to_main_menu_index = all_dict_values.index(' - Back to Main Menu')
                
            if number == str(back_to_main_menu_index + 1):
                return False
            
            selected_output_type = section.select_output_type()

            try:
                data_response = ''
                if number == '1':
                    data_response=te.getWBCategories(output_type=selected_output_type)

                if number == '2':
                    selected_indicator = section.select('indicatior','education')
                    data_response=te.getWBCategories(category=selected_indicator,output_type=selected_output_type)
                
                if number == '3':
                    selected_series_code = section.select('series code','SE.ADT.1524.LT.FE.ZS')
                    data_response=te.getWBIndicator(series_code = selected_series_code, output_type = selected_output_type)
                
                if number == '4':
                    selected_country = section.select('country','portugal')
                    data_response=te.getWBCountry(country = selected_country, output_type = selected_output_type)
                
                if number == '5':
                    selected_series_code = section.select('series code','usa.fr.inr.rinr')
                    data_response=te.getWBIndicator(series_code = selected_series_code, output_type = selected_output_type)
                
                if number == '6':
                    selected_url = section.select('url','/united-states/real-interest-rate-percent-wb-data.html')
                    data_response=te.getWBIndicator( url = selected_url, output_type = selected_output_type)

                if number == '7':
                    selected_series_code = section.select('series code','usa.fr.inr.rinr')
                    data_response=te.getWBHistorical(series_code =selected_series_code, output_type = selected_output_type)
                    
                if selected_output_type == 'df':
                    print(data_response.to_string())
                else:
                    print(data_response)

                return 'continue_or_back_to_main_menu'
            except Exception as e:
                print('**************************************************')
                print(e) 
                    
            
            return str(number)
        
        function_selected = True
        section = Section()
        while function_selected:
            function_selected=select_function(self)
            if function_selected == 'continue_or_back_to_main_menu':
                function_selected = section.new_request_or_back_to_menu(self.section_name)
                






         

        
        
        

    

    

    
    

    

        



    

    

